core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -155,7 +155,10 @@ public void close() throws IOException {
     }
     updateStreams();
     if (shouldCachePartiallyReadBlock()) {
-      readCurrentBlockToEnd();
+      // cache only when the read is not from local worker and there is a local worker to cache
+      if (!isReadFromLocalWorker() && !isRemoteReadButNoLocalWorker()) {
```

Comment:
We use this compound condition a few times, could you make it a method?

Reply:
hmm it's only used twice, and one of them is in conjunction with another condition. So I feel wrap it as a method is an overkill.

Reply:
I think we use it 3 places counting this one, I'm more concerned that the compound condition does not intuitively convey what we are trying to check.

Reply:
done